docs/explanation/e-hardening.md

```diff
@@ -0,0 +1,119 @@
+# Security Hardening Guide
+
+This document provides all the guidance and instructions needed to achieve 
+a secure deployment of the Charmed Kafka product, including setting up and managing a secure environment.
+The document is divided into the following sections:
+
+1. Environment, outlining the recommendation for deploying a secure environment
+2. Applications, outlining the product features that enable a secure deployment of a Kafka cluster
+3. Additional resources, providing any further information about security and compliance
+
+## Environment
+
+The environment where applications operate can be divided in three components:
+
+1. Cloud
+2. Juju 
+
+### Cloud
+
+Charmed Kafka can be deployed on top of several clouds and virtualization layers. 
```

Comment:
```suggestion
Charmed Kafka can be deployed on top of multiple different cloud platforms:
```

Reply:
are you sure that we should use tables as bullet points? I would have agreed with you if we had bullet points but since we use a table, it would seem more appropriate to use a sentence. Is there any discussion on a style guide about this?

Reply:
Are you referring to the usage of an introductory sentence with a colon before the table?
It is the usual way to do it, for example, see our [Style guide](https://docs.ubuntu.com/styleguide/en#spelling).
We are not using a table as bullet points (list) here, but they are similar in some ways.
Consider a list to be a one-dimensional table or a table to be a two-dimensional list =).